### PR TITLE
[CORE-11326] Update to React 18

### PR DESCRIPTION
**Pull request by eliasbraun**
https://jira.nebenan.de/browse/CORE-12563

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,11 +53,11 @@
     "eslint": "^7.6.0",
     "eslint-config-nebenan": "^3.0.0",
     "mocha": "^8.2.1",
-    "react": "^17.0.1",
+    "react": "^18.2.0",
```
Comment (rejas) [inline package.json:56]: are the react devDependencies needed at all? dont see any tests or other dev stuff using them explicitly....

Comment (eliasbraun) [reply package.json:56]: Hmm you're right, looking at the history it was used back in the days but I also cannot find any usage...

**Comment by spossner** [comment]
and here also upgrade to mocha 10 and doing an `npm update`?

**Review by spossner — APPROVED** [review]
passt (assuming the minor server change is really enough)

File: package.json [hunk 2]
```diff
@@ -52,12 +52,10 @@
     "chai": "^4.2.0",
     "eslint": "^7.6.0",
     "eslint-config-nebenan": "^3.0.0",
-    "mocha": "^8.2.1",
-    "react": "^17.0.1",
+    "mocha": "^10.2.0",
     "react-router": "^5.2.0"
   },
   "peerDependencies": {
-    "react": "^16.13.1 || ^17.0.1",
```
Comment (rejas) [inline package.json:60]: but why dont we keep the peerDependency? is that already in react-router?

Comment (eliasbraun) [reply package.json:60]: We removed that peerDependency!

Comment (eliasbraun) [reply package.json:60]: Or am I misunderstanding you here?

Comment (rejas) [reply package.json:60]: yes, why did you remove it?

Comment (rejas) [reply package.json:60]: it's a react package after all

Comment (eliasbraun) [reply package.json:60]: Aaaah I messed up peerDependencies and devDependencies ☕️ 
Sorry for the confusion, fixed it, good catch!

**Review by rejas — APPROVED** [review]

**Pull request merged**